external/fv3fit/tests/training/test_main.py

```diff
@@ -86,86 +105,135 @@ def mock_load_batches():
         yield magic_load_mock
 
 
+def call_main(
+    tmpdir, mock_load_batches, additional_variables, derived_output_variables,
+):
+    model_type = "DenseModel"
+    hyperparameters = fv3fit.DenseHyperparameters(depth=2, width=8)
+    config = get_config(
+        tmpdir,
+        additional_variables,
+        derived_output_variables,
+        model_type,
+        hyperparameters,
+    )
+    mock_load_batches.return_value = [config.mock_dataset for _ in range(6)]
+    all_variables = (
+        config.input_variables + config.output_variables + additional_variables
+    )
+    with mock.patch("fv3fit.DerivedModel") as MockDerivedModel:
+        MockDerivedModel.return_value = mock.MagicMock(
+            name="derived_model_return", spec=fv3fit.Predictor
+        )
+        fv3fit.train.main(config.args)
+    return CallArtifacts(
+        config.output_path,
+        all_variables,
+        MockDerivedModel,
+        config.input_variables,
+        config.output_variables,
+        hyperparameters,
+    )
+
+
 @pytest.mark.parametrize("additional_variables", [[], ["downward_shortwave"]])
-def test_main(
-    mock_dataset: xr.Dataset,
+@pytest.mark.parametrize("derived_output_variables", [[], ["downwelling_shortwave"]])
+def test_main_calls_load_batches_correctly(
     tmpdir,
     mock_load_batches: mock.MagicMock,
     mock_train_dense_model: mock.MagicMock,
     additional_variables: Sequence[str],
+    derived_output_variables: Sequence[str],
 ):
     """
     Test of fv3fit.train main function only, using mocks for training function
     and data loading.
     """
-    mock_load_batches.return_value = [mock_dataset for _ in range(6)]
-    args, input_variables, output_variables, hyperparameters, output_path = get_config(
-        tmpdir, mock_dataset, additional_variables, []
+    artifacts = call_main(
+        tmpdir, mock_load_batches, additional_variables, derived_output_variables,
     )
-    all_variables = input_variables + output_variables + additional_variables
-    fv3fit.train.main(args)
-    mock_load_batches.assert_called_with(variables=all_variables)
+    mock_load_batches.assert_called_with(variables=artifacts.all_variables)
     assert mock_load_batches.call_args_list[0] == mock_load_batches.call_args_list[1]
     assert mock_load_batches.call_count == 2
-    mock_train_dense_model.assert_called_once_with(
-        input_variables=input_variables,
-        output_variables=output_variables,
-        hyperparameters=hyperparameters,
-        train_batches=mock_load_batches.return_value,
-        validation_batches=mock_load_batches.return_value,
+
+
+@pytest.mark.parametrize("additional_variables", [[], ["downward_shortwave"]])
```

Comment:
Is the parameterization of `additional_variables` actually relevant to this test?

Reply:
Similar to above, it could be, particularly if a bug makes it so derived variables are only correctly applied if additional variables are/aren't set.